src/core/main.c

```diff
@@ -355,11 +378,11 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "systemd.confirm_spawn") && value) {
 
-                r = parse_boolean(value);
+                arg_confirm_spawn = mfree(arg_confirm_spawn);
+
+                r = parse_confirm_spawn(value, &arg_confirm_spawn);
                 if (r < 0)
-                        log_warning("Failed to parse confirm spawn switch %s. Ignoring.", value);
-                else
-                        arg_confirm_spawn = r;
+                        log_warning_errno(r, "Failed to parse confirm_spawn switch %s. Ignoring.", value);
```

Comment:
I know @poettering wanted it this way, but it's not better ;(
Actually with the previous code, we always ended up with a clean state — either atomic success, or atomic failure. Here we free the previous value, so in the end we are not ignoring the failed parsing but resetting to defaults. But let's leave it this way for now, it's not important at all.